src/System.Text.Formatting/System/Text/Parsing/InvariantParser_byte.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Diagnostics;
+using System.Text.Formatting;
+using System.Text.Utf8;
+
+namespace System.Text.Parsing
+{
+    public static partial class InvariantParser
+    {
+        /// <summary>
+        /// Parses an unsigned 64-bit integer from a location within a UTF-8 byte array buffer.
+        /// </summary>
+        /// <param name="utf8Text">The UTF-8 buffer, passed as a byte array.</param>
+        /// <param name="index">The index location of the value to be parsed within the buffer.</param>
+        /// <param name="value">The parsed value.</param>
+        /// <param name="bytesConsumed">The length (in bytes) of the unparsed value within the UTF-8 buffer.</param>
+        /// <returns>True if parsing is successful; false otherwise.</returns>
+		public static bool TryParse(byte[] utf8Text, int index, out byte value, out int bytesConsumed)
+        {
+            Precondition.Require(utf8Text.Length > 0);
+
+            value = 0;
+            bytesConsumed = 0;
+
+            for (int byteIndex = index; byteIndex < utf8Text.Length; byteIndex++) // loop through the byte array
+            {
+                byte nextByte = utf8Text[byteIndex];
+                if (nextByte < '0' || nextByte > '9') // if the next character is not a digit
```

Comment:
Utf-8 characters are 1 to 4 bytes long. If the value of a byte is greater than `0x7F` then you know that the byte cannot stand alone and you are in fact mid-sequence. In your case, you can likely just fail out of the test and ignore the byte.

I must have read this code while quite tired, as the `nextByte < '0'` will guard you against any issues cause by UTF-8 encoding.